crates/llvm-context/src/polkavm/context/mod.rs

```diff
@@ -669,20 +669,6 @@ where
                 self.build_byte_swap(value)
             }
             AddressSpace::Storage | AddressSpace::TransientStorage => {
-                let storage_key_value = self.builder().build_ptr_to_int(
-                    pointer.value,
-                    self.word_type(),
-                    "storage_ptr_to_int",
-                )?;
-                let storage_key_pointer = self.build_alloca(self.word_type(), "storage_key");
-                let storage_key_pointer_casted = self.builder().build_ptr_to_int(
-                    storage_key_pointer.value,
-                    self.xlen_type(),
-                    "storage_key_pointer_casted",
-                )?;
-                self.builder()
-                    .build_store(storage_key_pointer.value, storage_key_value)?;
-
                 let storage_value_pointer =
                     self.build_alloca(self.word_type(), "storage_value_pointer");
```

Comment:
```suggestion
                    self.build_alloca(self.word_type(), "storage_value_pointer");
                    self.build_store(storage_value_pointer, self.word_const(0))?;
```
It looks to me like this instead of emiting the branch later on would be more efficient code size wise. Can you try this and revert the change below that emits the branch to confirm less code size increase for the ERC20 and flipper contracts In the diff above?

Reply:
Thinking about it, adding that to the Storage.sol test fixture (reading a nonexistant key and asserting it returns a zero value) wouldn't harm.

Reply:
We will definitively check again when we have 64bit support, gas metering and everything. Also if either one is more efficient in either dimension (code size vs. execution performance), we can just emit either one based on the optimizations request (i.e. size optimization will emit this but not -O3).

Reply:
But by default we should optimize for size here. Storage operations are costly anyways.